app/controllers/admin/api/accounts_controller.rb

```diff
@@ -110,6 +115,12 @@ def buyer_account
     @buyer_account ||= buyer_accounts.find(params[:id])
   end
 
+  def to_present(accounts)
+    # ActiveRecord::Associations::Preloader.new(records: Array(accounts), associations: [:annotations, {bought_plans: %i[original]}]).call # Rails 7.x
+    ActiveRecord::Associations::Preloader.new.preload(Array(accounts), [:annotations, {bought_plans: %i[original]}])
```

Comment:
Is this method ever used for arrays? if not, maybe we should rename the argument as `account` to make it clear it's accepts and returns a single object?


Otherwise (to make it more generic), it can be left as it is, but then I guess it should be `Array(accounts).flatten` in case `accounts` is actually an array?

Reply:
oh OK, nice

Reply:
It's the first time I see `Preloader`, what's the advantage over `.includes`? Why is this method supposed to receive "accounts" but instead it's called with `buyer_account`?

Reply:
there are no advantages over `#includes`. Issue is that `#includes` only operates with Associations/queries that are not already performed. If your method receives a bunch of active record objects, then you can't call `#includes` on them.